src/utils/validation.ts

```diff
@@ -0,0 +1,15 @@
+export const validInputPattern = new RegExp(/^\d+\.?\d*$/) // allows leading and trailing zeros
+
+export function preventInvalidChars(event: React.KeyboardEvent<HTMLInputElement>): void {
+  if (
+    event.key !== 'Backspace' &&
+    event.key !== 'Delete' &&
```

Comment:
I wonder if these kind of guards can be tricked with `Ctrl+X` - cut text :thinking:

Reply:
Maybe, but in the end, `onBlur` should clear out that stuff